src/en/readcomic/src/eu/kanade/tachiyomi/extension/en/readcomic/ReadComic.kt

```diff
@@ -0,0 +1,226 @@
+package eu.kanade.tachiyomi.extension.en.readcomic
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.ArrayList
+import java.util.Date
+import java.util.Locale
+
+class ReadComic : ParsedHttpSource() {
+
+    override val name = "ReadComic"
+
+    override val baseUrl = "https://readcomic.top"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    override fun popularMangaSelector() = "div.eg-box"
+
+    override fun latestUpdatesSelector() = "ul.line-list"
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/popular-comics".toHttpUrl().newBuilder().apply {
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url.toString(), headers)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/comic-updates".toHttpUrl().newBuilder().apply {
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url.toString(), headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/advanced-search".toHttpUrl().newBuilder().apply {
+            addQueryParameter("key", query)
+            filters.forEach { filter ->
+                when (filter) {
+                    is GenreFilter -> addQueryParameter("wg", filter.toUriPart())
+                    else -> {}
+                }
+            }
+            if (page > 1) addQueryParameter("page", page.toString())
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.select("div.egb-right > a.egb-serie").attr("href"))
+        title = element.select("div.egb-right > a.egb-serie").text()
+        thumbnail_url = element.select("a.eg-image > img").attr("src")
+    }
+
+    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
+        val selector = "ul.line-list > li > a.big-link"
+        setUrlWithoutDomain(element.select(selector).attr("href"))
+        title = element.select(selector).text()
+        thumbnail_url = fetchThumbnailURL(element.select(selector).attr("href"))
```

Comment:
this will result in n calls for n comics, so dont.

you can put something like: `https://fakeimg.pl/200x300/?text=No%20Cover&font_size=62` as placeholder